test/arelx_test_helper.rb

```diff
@@ -6,6 +6,58 @@
 
 require 'support/fake_record'
 
+$colored = ENV["TERM"] =~ /^xterm|-256color$/ ? ->(color, msg){ "\x1b[#{color}m#{msg}\x1b[89m\x1b[0m" } : ->(color, msg){ "#{msg}" }
+YELLOW = "33"
+
+def warn(msg)
+  $stderr.puts($colored[YELLOW, msg])
+end
+
+# Load gems specific to databases
+# NOTE: It's strongly advised to test each database on its own.
+#       Loading multiple backend gems leads to undefined behavior according to
+#       tests; the backend might not recognize the correct DB visitor and will
+#       fallback to `ToSQL` and screw all tests.
+db_and_gem =  if RUBY_ENGINE == 'jruby'
+                {
+                  'mysql'      => 'activerecord-jdbcmysql-adapter',
+                  'postgresql' => 'activerecord-jdbcpostgresql-adapter',
+                  'sqlite'     => 'activerecord-jdbcsqlite3-adapter',
+                  'ibm_db'     => 'ibm_db',
+                  'oracle'     => 'activerecord-oracle_enhanced-adapter',
+                  'mssql'      => 'activerecord-jdbcsqlserver-adapter'
+                }
+              else
+                {
+                  'mysql'      => 'mysql2',
+                  'postgresql' => 'pg',
+                  'sqlite'     => 'sqlite3',
+                  'ibm_db'     => 'ibm_db',
+                  'oracle'     => 'activerecord-oracle_enhanced-adapter',
+                  'mssql'      => 'activerecord-sqlserver-adapter'
+                }
+              end
+
+dbs_to_test   = ENV['DB'].nil? || ENV['DB'].strip.empty? ? db_and_gem : db_and_gem.select{ |db, _| db == ENV['DB'] }
+
+if dbs_to_test.size > 1
+  warn "Warning: you're testing multiple databases at the same time:\n"
+  warn "    #{dbs_to_test.keys.to_a}\n"
+  warn "This can lead to undefined behavior and will fail (even error) otherwise perfectly correct tests.\n"
+  warn "Are you running `rake test`? Try running each db test separately, like `rake test:mysql`.\n"
+end
+
+def load_lib(gem)
+  begin
+    Gem::Specification.find_by_name(gem)
+    require gem
+  rescue Gem::MissingSpecError
+    warn "Warning: failed to load gem #{gem}. Are you sure it's installed?"
+  end
+end
+
+dbs_to_test.map { |_, gem| load_lib(gem)}
```

Comment:
Looks like an `each` to me.  You could inline load_lib in it.

Reply:
Actually this whole thing was a vestigial remain of something else. I am redoing it to something that makes sense.